main/src/ca/uwaterloo/flix/language/phase/Deriver.scala

```diff
@@ -539,6 +540,138 @@ object Deriver extends Phase[KindedAst.Root, KindedAst.Root] {
       KindedAst.MatchRule(pat, guard, exp)
   }
 
+  /**
+    * Creates a Hash instance for the given enum.
+    *
+    * {{{
+    * enum E[a] with Hash {
+    *   case C0
+    *   case C1(a)
+    *   case C2(a, a)
+    * }
+    * }}}
+    *
+    * yields
+    *
+    * {{{
+    * instance Hash[E[a]] with Hash[a] {
+    *   pub def hash(x: E[a]): Int = match x {
+    *     case C0 => 0
```

Comment:
I also wonder if it is a good idea to number the cases with 0, 1, and so forth.

Reply:
I thought it wouldn't be a problem since the hash code of any integer type is just itself.

Reply:
OK